### PR TITLE
Throw when trying to send data that doesn't fit into the MTU.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/resources/ble_esp32.cc [hunk 1]
```diff
@@ -868,6 +874,17 @@ BleCharacteristicResource::~BleCharacteristicResource() {
   service_->remove_characteristic(this);
 }
 
+uint16 BleCharacteristicResource::get_mtu() {
+  int min_sub_mtu = -1;
```
Comment (floitsch) [inline src/resources/ble_esp32.cc:878]: Code that was copied from the `get_att_mtu` primitive.
Tbh, I don't really understand it.

File: src/resources/ble_esp32.cc [hunk 2]
```diff
@@ -1343,24 +1360,28 @@ int BlePeripheralManagerResource::_on_gap(const BleCallbackScope& scope, struct
   return BLE_ERR_SUCCESS;
 }
 
-static Object* object_to_mbuf(Process* process, Object* object, os_mbuf** result) {
```
Comment (floitsch) [inline src/resources/ble_esp32.cc:1346]: Split into two functions on the right. One that takes an object and one that takes a blob.

File: src/resources/ble_esp32.cc [hunk 3]
```diff
@@ -1894,14 +1915,23 @@ PRIMITIVE(get_value) {
...
 
 PRIMITIVE(write_value) {
-  ARGS(BleReadWriteElement, element, Object, value, bool, with_response, bool, flush, bool, allow_retry)
+  ARGS(BleReadWriteElement, element, Blob, value, bool, with_response, bool, flush, bool, allow_retry)
 
   Locker locker(BleResourceGroup::instance()->mutex());
 
   if (!element->service()->device()) FAIL(INVALID_ARGUMENT);
 
+  uint16 mtu;
```
Comment (kasperl) [inline src/resources/ble_esp32.cc:1924]: Not sure it is really beneficial to have this as a 16 bit local, but it might be the style around here.

**Review by kasperl — COMMENTED** [review]
Looks good to me.

**Review by kasperl — APPROVED** [review]

File: src/resources/ble_esp32.cc (hunk 3 above)
Comment (floitsch) [reply src/resources/ble_esp32.cc]: done.

**Pull request merged**